src/omnisharp/experimentalOmnisharp.Downloader.ts

```diff
@@ -0,0 +1,72 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { PackageManager, Package } from '../packages';
+import { PlatformInformation } from '../platform';
+import { Logger } from '../logger';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { GetPackagesFromVersion } from './experimentalOmnisharp.PackageCreator';
+import { GetDependenciesAndDownloadPackages, GetStatus, GetAndLogPlatformInformation, ReportInstallationError, SendInstallationTelemetry } from '../experimentalOmnisharp.Download.Helper';
+
+export class ExperimentalOmnisharpDownloader {
+
+    public constructor(
+        private channel: vscode.OutputChannel,
+        private logger: Logger,
+        private reporter: TelemetryReporter /* optional */,
+        private packageJSON: any) {
+    }
+
+    public async DownloadAndInstallExperimentalVersion(version: string, serverUrl: string, installPath: string) {
+        if (!version) {
+            throw new Error('Invalid version');
+        }
+
+        this.logger.append('Getting the version packages...');
+        this.logger.appendLine();
+        this.channel.show();
+
+        let statusItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right);
```

Comment:
The pattern here is: "call raw VSCode APIs to add status bar item, then call a helper to wrap the VSCode status item into something easy to manipulate". Can we push the VSCode API call into the "GetStatus" helper (and call it something like "SetStatus")?